From: Ben Hutchings <ben@decadent.org.uk>
Date: Wed, 15 Jan 2014 02:09:13 +0000
Subject: mm: Avoid ABI change in 3.12.7
Forwarded: not-needed

Move the added member mm_struct::tlb_flush_pending to the end of
the structure and hide it from genksyms.

The structure is always allocated in the core kernel.

--- a/include/linux/mm_types.h
+++ b/include/linux/mm_types.h
@@ -437,6 +437,8 @@ struct mm_struct {
 	 */
 	int first_nid;
 #endif
+	struct uprobes_state uprobes_state;
+#ifndef __GENKSYMS__
 #if defined(CONFIG_NUMA_BALANCING) || defined(CONFIG_COMPACTION)
 	/*
 	 * An operation with batched TLB flushing is going on. Anything that
@@ -445,7 +447,7 @@ struct mm_struct {
 	 */
 	bool tlb_flush_pending;
 #endif
-	struct uprobes_state uprobes_state;
+#endif
 };
 
 /* first nid will either be a valid NID or one of these values */
